### DISCUSSION TITLE
How would I compile raylib for OpenGL 3.1

### DISCUSSION THREAD

**Discussion by Cube5x59999** [raysan5/raylib#4287] [Q&A]
I saw the wiki to compile for another versions on opengl and it says using notepad++, so I installed it and the wiki has no like which file to open first, so I'm stuck, does it mean compiling my program or which main library file?

I use vscode for my program

**Comment by CrackedPixel** [comment]
To change OpenGL versions you would want to change the  `GRAPHICS` flag in the Makefile (or config/etc depending on how you build raylib)

But 3.1 does not seem like a valid selection. The following are choices:
`GRAPHICS_API_OPENGL_11`
`GRAPHICS_API_OPENGL_21`
`GRAPHICS_API_OPENGL_33` (default)
`GRAPHICS_API_OPENGL_43`
`GRAPHICS_API_OPENGL_ES2` (mobile)


taken from https://github.com/raysan5/raylib/blob/master/src/Makefile#L235
